File: tatala-client-csharp/src/QiLeYuan/Tatala/socket/to/helper/WrapperMapHelper.cs
using System;
using System.Collections.Generic;
using QiLeYuan.Tatala.socket.io;
using QiLeYuan.Tatala.socket.util;

namespace QiLeYuan.Tatala.socket.to.helper {

    public class WrapperMapHelper {
        public static int getByteArrayLength(Dictionary<String, TransferObjectWrapper> wrapperMap) {
            int blength = 0;
            foreach (KeyValuePair<String, TransferObjectWrapper> item in wrapperMap) {
                // type
                blength += TransferUtil.getLengthOfByte();
                // key
                String key = item.Key;
                blength += TransferUtil.getLengthOfString(key);
                // value
                TransferObjectWrapper value = item.Value;
                blength += TransferUtil.getLengthOfWrapper(value);
            }
            return blength;
        }

        public static void getByteArray(Dictionary<String, TransferObjectWrapper> wrapperMap, TransferOutputStream touts) {
            foreach (KeyValuePair<String, TransferObjectWrapper> item in wrapperMap) {
                // type
                touts.writeByte(TransferObject.DATATYPE_WRAPPER);
                // key
                String key = item.Key;
                touts.writeString(key);
                // value
                TransferObjectWrapper value = item.Value;
                touts.writeWrapper(value);
            }
        }
    }
}

File: history/0.1.2/tatala-java/src/com/qileyuan/tatala/socket/to/helper/WrapperMapHelper.java
package com.qileyuan.tatala.socket.to.helper;

import java.util.Iterator;
import java.util.Map;

import com.qileyuan.tatala.socket.io.TransferOutputStream;
import com.qileyuan.tatala.socket.to.TransferObject;
import com.qileyuan.tatala.socket.to.TransferObjectWrapper;
import com.qileyuan.tatala.socket.util.TransferUtil;

public class WrapperMapHelper {
	public static int getByteArrayLength(Map<String, TransferObjectWrapper> wrapperMap){
		int blength = 0;
		Iterator<String> keyIt = wrapperMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			blength += TransferUtil.getLengthOfByte();
			// key
			String key = (String) keyIt.next();
			blength += TransferUtil.getLengthOfString(key);
			// value
			TransferObjectWrapper value = wrapperMap.get(key);
			blength += TransferUtil.getLengthOfWrapper(value);
		}
		return blength;
	}
	
	public static void getByteArray(Map<String, TransferObjectWrapper> wrapperMap, TransferOutputStream touts) {
		Iterator<String> keyIt = wrapperMap.keySet().iterator();
		while (keyIt.hasNext()) {
			// type
			touts.writeByte(TransferObject.DATATYPE_WRAPPER);
			// key
			String key = (String) keyIt.next();
			touts.writeString(key);
			// value
			TransferObjectWrapper value = wrapperMap.get(key);
			touts.writeWrapper(value);
		}
	}
}
